### PR TITLE
[FIX JENKINS-40924] - freestyle queue busted

### PR DESCRIPTION
**Pull request by kzantow**
# Description

To test: set executors to 0, add a freestyle project, run it from BO. It should show up. Stopping it before it has executed should also remove it.

Testing for this is best served in ATH.

See [JENKINS-40924](https://issues.jenkins-ci.org/browse/JENKINS-40924).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [x] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
:bee: works for me! nice tough one.

**Comment by i386** [comment]
Nice to see this fixed

fiix5sc

**Comment by michaelneale** [comment]
@kzantow test failures (mocha) again (odd, I thought this was fine befor ethe merge).

**Comment by michaelneale** [comment]
```
validating dependencies in /home/ubuntu/slave/workspace/blueocean_PR-801-GJFQKZT2DHBITY4UEBAK6BQHDYFBZPHNSIPPLU72M4PTC7FWWM6A/blueocean-core-js
@jenkins-cd/design-language should be 0.0.110 but found 0.0.112
```

**Comment by kzantow** [comment]
yes, realized this... trying to fix it. unprecedented. :)

**Comment by kzantow** [comment]
literally wasted at least 8 hours dealing with this mess.

**Comment by michaelneale** [comment]
:bee: checked it locally - and I think it passed the ATH (it got artifacts from the most recent build).

File: blueocean-core-js/package.json [hunk 1]
```diff
@@ -43,7 +44,6 @@
     "jsonwebtoken": "7.1.9",
     "mobx": "2.6.0",
     "mobx-utils": "1.1.2",
-    "moment": "2.15.1",
```
Comment (michaelneale) [inline blueocean-core-js/package.json]: how come this one is removed?

**Comment by michaelneale** [comment]
@scherler @cliffmeyers  HELP! Keith is trying to get shrinkcrap to update to 0.0.112, but it won't. The package.json is set but shitcrap says 0.0.110: https://github.com/kzantow/blueocean-plugin/blob/b55e88e835e8f58a735f2110c0cb66138ed2f6ed/blueocean-dashboard/npm-shrinkwrap.json no matter what he tries. WTF is going ON????

**Comment by michaelneale** [comment]
actually @kzantow - I am not sure if this really is happening how you think it is happening. 

Anytime I see a shrinkwrap.json with 0.0.110 - it is a dependency inside core-js. However, at one level up, there is 0.0.112. I wonder if this is just how it works - it resolves to the highest one? 

Eg compare: 

https://github.com/kzantow/blueocean-plugin/blob/b55e88e835e8f58a735f2110c0cb66138ed2f6ed/blueocean-dashboard/npm-shrinkwrap.json#L11

with: 

https://github.com/kzantow/blueocean-plugin/blob/b55e88e835e8f58a735f2110c0cb66138ed2f6ed/blueocean-dashboard/npm-shrinkwrap.json#L18

Is it actually working correctly because of this? 

(cc @scherler @cliffmeyers)

**Pull request merged**